src/gluonts/model/tft/_estimator.py

```diff
@@ -155,161 +155,106 @@ def __init__(
         )
 
     def create_transformation(self) -> Transformation:
-        transforms = (
-            [AsNumpyArray(field=FieldName.TARGET, expected_ndim=1)]
-            + (
-                [
-                    AsNumpyArray(field=name, expected_ndim=1)
-                    for name in self.static_cardinalities.keys()
-                ]
-            )
-            + [
-                AsNumpyArray(field=name, expected_ndim=1)
-                for name in chain(
-                    self.static_feature_dims.keys(),
-                    self.dynamic_cardinalities.keys(),
-                )
-            ]
-            + [
-                AsNumpyArray(field=name, expected_ndim=2)
-                for name in self.dynamic_feature_dims.keys()
-            ]
-            + [
-                AddObservedValuesIndicator(
-                    target_field=FieldName.TARGET,
-                    output_field=FieldName.OBSERVED_VALUES,
-                ),
-                AddTimeFeatures(
-                    start_field=FieldName.START,
-                    target_field=FieldName.TARGET,
-                    output_field=FieldName.FEAT_TIME,
-                    time_features=self.time_features,
-                    pred_length=self.prediction_length,
-                ),
-            ]
+        transform = AsNumpyArray(field=FieldName.TARGET, expected_ndim=1)
+
+        for name in self.static_cardinalities.keys():
+            transform += AsNumpyArray(field=name, expected_ndim=1)
+
+        for name in chain(
+            self.static_feature_dims.keys(),
+            self.dynamic_cardinalities.keys(),
+        ):
+            transform += AsNumpyArray(field=name, expected_ndim=1)
+
+        for name in self.dynamic_feature_dims.keys():
+            transform += AsNumpyArray(field=name, expected_ndim=2)
```

Comment:
Why is `expected_ndim` set to `2` here? Could we maybe group transforms with `expected_ndim = 1` and `expected_ndim = 2` and give it a more descriptive name?

Reply:
Because that's the expected number of dimensions of the array 😉 we expect lists-of-lists or 2-dimensional arrays whenever there's an additional axis other than time. In this case, you expect potentially multiple time-varying features